service/watcher/chartstatus/dynclient.go

```diff
@@ -76,7 +83,8 @@ func (c *ChartStatusWatcher) waitForAvailableConnection(ctx context.Context, dyn
 		c.logger.Debugf(ctx, "failed to get available connection: %#v retrying in %s", err, t)
 	}
 
-	b := backoff.NewExponential(5*time.Minute, 30*time.Second)
+	// maxWait is 0 since cluster creation may fail.
+	b := backoff.NewExponential(0, 30*time.Second)
```

Comment:
>	// After MaxElapsedTime the ExponentialBackOff returns Stop.
>	// It never stops if MaxElapsedTime == 0.

https://github.com/cenkalti/backoff/blob/6b0e4ad0cd65431b217393bff47d1dff727b264b/exponential.go#L59_L
61

Since cluster creation takes more than 5 mins the backoff then returns Stop and retries after 0 seconds. 🤦‍♂️ 

If there is a problem the cluster may never come up. So we should keep retrying every 30 secs.